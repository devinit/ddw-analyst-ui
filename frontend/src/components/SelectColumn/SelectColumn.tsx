import React, { FunctionComponent } from 'react';
import { useSortable } from '@dnd-kit/sortable';
import { CSS } from '@dnd-kit/utilities';
import styled from 'styled-components';

interface SelectColumnProps {
  id: string;
}

const Column = styled.div`
  border-style: solid;
  width: 400px;
  margin: 2px;
  cursor: grab;
  padding: 5px;
  border-color: grey;
`;

const SelectColumn: FunctionComponent<SelectColumnProps> = (props) => {
  const { attributes, listeners, setNodeRef, transform, transition } = useSortable({
    id: props.id,
  });

  const style = {
    transform: CSS.Transform.toString(transform),
    transition: transition?.toString(),
  };

  return (
    <Column style={style} ref={setNodeRef} {...attributes} {...listeners}>
      {props.id}
    </Column>
  );
};

export { SelectColumn };
