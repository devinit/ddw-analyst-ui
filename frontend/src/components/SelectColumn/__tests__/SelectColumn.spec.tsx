import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { SelectColumn } from '../SelectColumn';

test('renders correctly with the required props', () => {
  const renderer = TestRenderer.create(<SelectColumn id="test" />).toJSON();

  expect(renderer).toMatchSnapshot();
});
