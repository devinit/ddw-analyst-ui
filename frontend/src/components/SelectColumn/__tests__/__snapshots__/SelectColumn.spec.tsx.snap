// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with the required props 1`] = `
<div
  aria-describedby=""
  aria-roledescription="sortable"
  className="sc-bdfBwQ gWmvCG"
  role="button"
  style={
    Object {
      "transform": undefined,
      "transition": undefined,
    }
  }
  tabIndex={0}
>
  test
</div>
`;
